homeassistant/components/yale_smart_alarm/coordinator.py

```diff
@@ -137,3 +162,17 @@ def get_updates(self) -> dict:
             "status": status,
             "online": online,
         }
+
+    def auth_failed(self, error) -> None:
```

Comment:
You could skip this whole function and instead `raise ConfigEntryAuthFailed from error` which is handled by DataUpdateCoordinator.  It will log the problem and start reauth for you.  See update_coordinator.py